draft-patton-cfrg-vdaf.md

```diff
@@ -705,6 +705,159 @@ TODO
 * Output shares are secret shares of a vector of field elements, each
   corresponding to a counter for one of the candidate prefixes.
 
+## Dependencies
+
+[TODO Clarify which of these can be assumed and which need to be explicitly
+defined]
+
+### Descriptions of finite fields
+
+We assume a structure for describing IDPF output groups, named `ValueType`.
+For verifiable heavy hitters, we need an IDPF that at each level has two field
+elements as outputs, and so the value type would be pairs of field elements.
+The IDPF key generation and evaluation functions will allow a different `ValueType`
+parameter per bit in the IDPF domain, which allows us to use different types at
+the output layer, required for extractability. See [BBCGGI21, Section 4.3] for
+details. We assume a function `get_value_type(n: Unsigned) -> ValueType` that
+returns the type for bit index `n`. We use `ValueType` to refer to a description
+of a given group, and`Value` to refer to an actual element of the group.
+
+### Incremental Distributed Point Functions (IDPFs)
+
+An IDPF is defined over a domain of size `2**d`. The client can specify an
+index `alpha` and values `beta`, one for each level `l` in `[d]`. The key
+generation generates two IDPF keys that individually hide `alpha` and `beta`.
+When locally evaluated at any point `x` in `2**l` at level `l`, the IDPF
+returns shares of `beta[l]`, if `x` is the `l`-bit prefix of `alpha`, and
+shares of zero otherwise. If elements of `beta` are tuples, the additive
+sharing is done element-wise.
+
+- `idpf_gen(alpha: Unsigned, beta: Vec[Value]) -> (Bytes, Bytes)`:
```

Comment:
I've been naming the output parameters. How about `-> (idpf_key1: Bytes, idpf_key2: Bytes)` or something?